package/Shaders/Lighting.hlsl

```diff
@@ -1853,17 +2261,44 @@ PS_OUTPUT main(PS_INPUT input, bool frontFace
 #		endif
 #	endif
 
-	float4 color;
-	color.xyz = diffuseColor * baseColor.xyz;
-
 #	if defined(HAIR)
-	float3 vertexColor = (input.Color.yyy * (TintColor.xyz - 1.0.xxx) + 1.0.xxx);
+	float3 vertexColor = lerp(1, TintColor.xyz, input.Color.y);
 #	else
 	float3 vertexColor = input.Color.xyz;
 #	endif  // defined (HAIR)
-	float3 realVertexColor = vertexColor;
 
-	vertexColor *= color.xyz;
+	float4 color = 0;
+
+#	if defined(TRUE_PBR)
+	{
+		float3 directLightsDiffuseInput = diffuseColor * baseColor.xyz;
+		[branch] if ((PBRFlags & TruePBR_ColoredCoat) != 0)
+		{
+			directLightsDiffuseInput = lerp(directLightsDiffuseInput, pbrSurfaceProperties.CoatColor * coatLightsDiffuseColor, pbrSurfaceProperties.CoatStrength);
+		}
+
+		color.xyz += directLightsDiffuseInput;
+	}
+
+	float3 indirectDiffuseLobeWeight, indirectSpecularLobeWeight;
+	GetPBRIndirectLobeWeights(indirectDiffuseLobeWeight, indirectSpecularLobeWeight, worldSpaceNormal.xyz, worldSpaceViewDirection, baseColor.xyz, pbrSurfaceProperties);
+
+#		if !defined(DEFERRED)
+#			if !defined(SSGI)
+	color.xyz += indirectDiffuseLobeWeight * directionalAmbientColor;
+#			endif
+	specularColorPBR += indirectSpecularLobeWeight * GetDynamicCubemapSpecularIrradiance(screenUV, worldSpaceNormal, worldSpaceVertexNormal, worldSpaceViewDirection, pbrSurfaceProperties.Roughness, viewPosition.z);
```

Comment:
Does not compile in permutations that don't have DYNAMIC_CUBEMAPS

Reply:
Fixed